perceval/backends/stackexchange.py

```diff
@@ -0,0 +1,296 @@
+
+        cache_items = self.cache.retrieve()
+
+        for question in cache_items:
+            yield question
+
+
+class StackExchangeClient:
+    """StackExchange API client.
+
+    This class implements a simple client to retrieve questions from
+    any Stackexchange site.
+
+    :param site: URL of the Bugzilla server
+    :param tagged: filter items by question Tag
+    :param token: StackExchange access_token for the API
+    :param max_questions: max number of questions per query
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    def __init__(self, site, tagged, token, max_questions):
+        self.site = site
+        self.tagged = tagged
+        self.token = token
+        self.max_questions = max_questions
+        self.version = VERSION_API
+
+    def __build_base_url(self, type='questions'):
+        base_api_url = STACKEXCHANGE_API_URL
+        base_api_url = urljoin(base_api_url, self.version, type)
+        return base_api_url
+
+    def __build_payload(self, page, from_date, order='desc', sort='activity'):
+        payload = {'page': page,
+                   'pagesize': self.max_questions,
+                   'order': order,
+                   'sort': sort,
+                   'tagged': self.tagged,
+                   'site': self.site,
+                   'key': self.token,
+                   'filter': QUESTIONS_FILTER}
+        if from_date:
+            timestamp = int(time.mktime(from_date.timetuple()))
+            payload['min'] = timestamp
+        return payload
+
+    def __log_status(self, quota_remaining, quota_max, page_size, total):
+
+        logger.info("Rate limit: %s/%s" % (quota_remaining,
+                                           quota_max))
+        if (total != 0):
+            if (total <= page_size):
+                logger.info("Fetching questions: %s/%s" % (total,
+                                                           total))
+            else:
+                logger.info("Fetching questions: %s/%s" % (page_size,
+                                                           total))
+        else:
+            logger.info("No questions were found.")
+
+    def get_questions(self, from_date):
+        """Retrieve all the questions from a given date.
+
+        :param from_date: obtain questions updated since this date
+        """
+
+        page = 1
+        req = requests.get(self.__build_base_url(),
+                           params=self.__build_payload(page, from_date))
+        req.raise_for_status()
+        questions = req.json()['items']
+
+        if (req.json()['page_size'] >= req.json()['total']):
+            fetched = req.json()['total']
+        else:
+            fetched = req.json()['page_size']
+
+        self.__log_status(req.json()['quota_remaining'],
+                          req.json()['quota_max'],
+                          fetched,
+                          req.json()['total'])
+
+        while questions:
```

Comment:
Ok. After all talk, you're right that it has more sense to return all the items related to a query because modern APIs allow us to do so in an easy way. The only thing here is you have to return the raw response with the `yield` statement:

```python
req = requests.get(self.__build_base_url(),
                   params=self.__build_payload(page, from_date))
questions = req.text

while questions:
    yield questions
    
    data = req.json()
    if data['has_more']:
        # do the next requests
       [...]
```